responsibleai/responsibleai/managers/explainer_manager.py

```diff
@@ -141,7 +142,18 @@ def _compute_explanations(self, local: bool, data: Any):
             model_task=model_task,
             classes=self._classes,
             categorical_features=self._categorical_features)
-        return explainer.explain_global(data, include_local=local)
+
+        if len(data) <= MAXIMUM_ROWS_FOR_GLOBAL_EXPLANATIONS:
+            return explainer.explain_global(data, include_local=local)
+        else:
+            warnings.warn((
+                "LARGE-DATA-SCENARIO-DETECTED: "
+                "The data is larger than the supported limit of {0}. "
+                "Computing explanations for first {0} samples only.").format(
+                    MAXIMUM_ROWS_FOR_GLOBAL_EXPLANATIONS),
+                UserWarning)
+            return explainer.explain_global(data[
```

Comment:
can we choose the points randomly instead of just the first 5k?  Sometimes data is ordered by some feature or label and this may give skewed results for the global explanation.

Reply:
Great idea. I plan to add sampling of data for another requirement and will change the above logic to perform random sampling based of that in a follow up PR.